api/v1alpha1/kubeadmbootstrapconfig_types.go

```diff
@@ -18,21 +18,28 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeadmv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Phase defines KubeadmBootstrapConfig phases
+type Phase string
+
+// Ready defines the KubeadmBootstrapConfig Ready Phase
+const Ready Phase = "Ready"
 
 // KubeadmBootstrapConfigSpec defines the desired state of KubeadmBootstrapConfig
 type KubeadmBootstrapConfigSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	ClusterConfiguration kubeadmv1beta1.ClusterConfiguration `json:"clusterConfiguration"`
+	InitConfiguration    kubeadmv1beta1.InitConfiguration    `json:"initConfiguration,omitempty"`
+	JoinConfiguration    kubeadmv1beta1.JoinConfiguration    `json:"joinConfiguration,omitempty"`
```

Comment:
@chuckha 
how are kube-proxy knobs handled?

Reply:
How does kubeadm handle the kube-proxy knobs?

Reply:
KubeletConfiguration and KubeProxyConfiguration can be passed as part of the same config.yaml, separated by `---`:
https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2

Reply:
in that case we may need to expand the Fields listed here -- I don't know if we're ready for exclusively v1b2 since we are supporting down to 1.13.

Reply:
the multi-doc support with the component configs (proxy/kubelet) came in v1alpha3 i think.
but yes v1beta1 will be removed in 1.18 so consuming projects can stay on it for a while.

Reply:
How can we handle the knobs using v1beta1 components if we don't have the KubeletConfiguration and KubeProxyConfiguration?

Reply:
you can still expose the public types:
https://github.com/kubernetes/kube-proxy
https://github.com/kubernetes/kubelet
(under config for these repos)